napari/_vispy/_tests/test_vispy_labels_layer.py

```diff
@@ -126,3 +126,13 @@ def test_labels_iso_gradient_modes(qtbot, qt_viewer):
     QCoreApplication.instance().processEvents()
     assert layer.iso_gradient_mode == 'fast'
     assert visual.node.iso_gradient_mode == 'fast'
+
+
+def test_int8_labels_color(qtbot, qt_viewer):
+    """Check that int8 labels colors can be shuffled without overflow.
+
+    See https://github.com/napari/napari/issues/7277.
+    """
+    data = np.arange(-128, 128, dtype=np.int8).reshape((16, 16))
+    layer = qt_viewer.viewer.add_labels(data)
```

Comment:
Why qt_viewer, why not just create Labels layer?

Reply:
good point! I wasn't 100% sure whether I needed VispyLabels to get instantiated to see the effect, but I just checked and I can indeed cause the failure with just labels. Will push a smaller test in test_labels.py

Reply:
Fixed now. Will merge if CI passes. (new test passed locally, and fails in main)